File: src/main/java/nl/hsleiden/model/Basket.java
package nl.hsleiden.model;

public class Basket {
    private int basketID;
    private int basketUserID;
    private int basketItemID;
    private int basketItemAmount;

    public Basket() {

    }

    public Basket(int basketID, int basketUserID, int basketItemID, int basketItemAmount) {
        this.basketID = basketID;
        this.basketUserID = basketUserID;
        this.basketItemID = basketItemID;
        this.basketItemAmount = basketItemAmount;
    }

    public int getBasketID() {
        return basketID;
    }

    public void setBasketID(int basketID) {
        this.basketID = basketID;
    }

    public int getBasketUserID() {
        return basketUserID;
    }

    public void setBasketUserID(int basketUserID) {
        this.basketUserID = basketUserID;
    }

    public int getBasketItemID() {
        return basketItemID;
    }

    public void setBasketItemID(int basketItemID) {
        this.basketItemID = basketItemID;
    }

    public int getBasketItemAmount() {
        return basketItemAmount;
    }

    public void setBasketItemAmount(int basketItemAmount) {
        this.basketItemAmount = basketItemAmount;
    }
}

File: webshop/src/app/shared/modelsAndTheirServices/basket.ts
export class Basket {

  private basketID;
  private basketUserID;
  private basketItemID;
  private basketItemAmount;

  constructor(private data?) {
    this.basketID = data.basketID;
    this.basketUserID = data.basketUserID;
    this.basketItemID = data.basketItemID;
    this.basketItemAmount = data.basketItemAmount;
  }

  public getData() {
    const data = {
      basketID: this.basketID,
      basketUserID: this.basketUserID,
      basketItemID: this.basketItemID,
      basketItemAmount: this.basketItemAmount
    };
    return data;
  }

  public getBasketID() {
    return this.basketID;
  }

  public getBasketUserID() {
    return this.basketUserID;
  }

  public getBasketItemID() {
    return this.basketItemID;
  }

  public getBasketItemAmount() {
    return this.basketItemAmount;
  }

  public addAnotherOne() {
    this.basketItemAmount++;
  }
}
